src/Infrastructure/Services/BindleService.cs

```diff
@@ -13,4 +13,12 @@ public BindleService(IConfiguration configuration)
         var bindleUrl = configuration.GetValue<string>("Bindle:Url", "http://127.0.0.1:8080/v1");
         _client = new BindleClient(bindleUrl);
     }
+
+    public async Task<IEnumerable<string?>> QueryAvailableStorages(string query, ulong? offset, int? limit)
+    {
+        var matches = await _client.QueryInvoices(query, offset, limit);
+        if (matches.Total == 0)
+            return new List<string?>();
+        return matches.Invoices.Select(i => i.Bindle?.Name).Distinct();
```

Comment:
Hmm... Perhaps we should not select invoices with a null bindle to avoid handling null strings?

I'm pretty sure there's never a case where an invoice would return a null bindle in the response body. Perhaps we should ignore and/or discard it as a possible case.